File: api/python/insert_executemany/insert_data.py
#!/usr/bin/env python3
# -*- coding: UTF-8 -*-

import sqlite3

conn = sqlite3.connect('test.db')
cur = conn.cursor()
print ("数据库打开成功")

employee_list = [
        (1, 'Paul', 32, 'California', 20000.00 ),
        (2, 'Allen', 25, 'Texas', 15000.00 ),
        (3, 'Teddy', 23, 'Norway', 20000.00 ),
        (4, 'Mark', 25, 'Rich-Mond ', 65000.00 ),
        ]

cur.executemany("INSERT INTO COMPANY (ID,NAME,AGE,ADDRESS,SALARY) VALUES (?, ?, ?, ?, ?)", employee_list)

conn.commit()
print ("数据插入成功")
conn.close()

File: api/cxx/insert/recipe-03/insert_data.cpp
#include <cstdio>
#include <SQLiteCpp/SQLiteCpp.h>
#include <SQLiteCpp/ExecuteMany.h>

int main()
{
    SQLite::Database db("test.db", SQLite::OPEN_READWRITE);
    printf("数据库打开成功\n");

    execute_many(db, 
            "INSERT INTO COMPANY (ID,NAME,AGE,ADDRESS,SALARY) VALUES (?, ?, ?, ?, ?)",
            std::make_tuple(1, "Paul", 32, "California", 20000.00 ),
            std::make_tuple(2, "Allen", 25, "Texas", 15000.00 ),
            std::make_tuple(3, "Teddy", 23, "Norway", 20000.00 ),
            std::make_tuple(4, "Mark", 25, "Rich-Mond ", 65000.00 ));

    printf("数据插入成功\n");

    return 0;
}

